shelley-mc/formal-spec/transactions.tex

```diff
@@ -118,65 +39,141 @@ \section{Transactions}
-      & \uniondistinct&
-        \type{RequireAll}~\powerset{\ScriptMPS}
-    \end{array}
-  \end{equation*}
-  \caption{The Multicurrency Scripting Language}
-  \label{fig:defs:tx-mc-script}
-\end{figure*}
+\begin{itemize}
+
+  \item $\Language$ is an abstract type that labels the scripting language, e.g.
+    $\mathsf{nativeMSigTag}$ or $\mathsf{nativeMCTag}$. It is kept abstract for easier extensibility in the future.
+
+  \item $\PolicyID$ identifies a specific currency.  As in normal life, two
+    currencies may use the same \emph{coinage} (sets of tokens that distinguish different values in the currency),
+    but are distinguished by their \emph{monetary policy}, which governs how tokens of the currency may be created
+    and destroyed.
+    The monetary policy for a currency $\var{pid}$ is given by the \emph{Monetary Policy Script}
+    (MPS) $s$, where $\fun{hashScript}~s~=~pid$. When a transaction attempts to create or destroy tokens,
+    the script verifies that the coinage for currency $\var{pid}$
+    respects the restrictions that are imposed by the
+    monetary policy. If the restrictions are complied with, it returns
+    $\mathsf{True}$, and
+    $\mathsf{False}$ otherwise.  Monetary Policy Scripts are described in more detail below.
+
+%% \item $\PolicyID$ identifies a specific monetary policy.  Each monetary policy is distinct. The monetary
+%%   policy for a currency $\var{c} = \{ \var{tok_1} \ldots \var{tok}_n \}$ is given by the \emph{Monetary Policy Script}
+%%   (MPS) $s$, such that $\fun{hashScript}~s~=~pid$. The script is called a
+%%   monetary policy script because when run, it verifies that the tokens of a currency
+%%   with the corresponding $\var{pid}$ are according to the restrictions imposed by the
+%%   monetary policy. If the restrictions are complied with, it returns
+%%   $\mathsf{True}$, and
+%%   $\mathsf{False}$ otherwise.  Monetary Policy Scripts are described in more detail below.
+
+
+  \item $\AssetID$ : the coinage for a currency $\var{pid} \in \var{PolicyID}$ is a set of terms
+    $t\in\AssetID$.  Each $t$ identifies a unique kind of token in $\var{pid}$.
+    We will assume that the token for Ada is $(\mathsf{adaID}, \mathsf{adaToken})$.
+
+  \item $\Quantity$ is an integer type that represents an amount of a specific $\AssetID$. We associate
+    a term $q\in\Quantity$ with a specific token to track how much of that token is contained in a given currency value.
+
+  \item $\Value$ is the multicurrency type that is used to represent
+    an amount of a collection of tokens, including Ada. If $(\var{pid}, \var{tok})$ is a token and $v \in \Value$,
+    the amount in $v$ belonging to that token is $v \var{pid} \var{tok}$ if defined, or zero otherwise.
+    Token amounts are fungible with each other if and only if they belong to the same token,
+    i.e. they have the same $\PolicyID$ and $\AssetID$.
+
```

Comment:
make a note here like "we refer to terms of type Value as "token bundles"".  This term is commonly used but not mentioned anywhere here i think